quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/PropertyMappingInterceptor.java

```diff
@@ -105,34 +108,45 @@ public Iterator<String> iterateNames(ConfigSourceInterceptorContext context) {
             }
             allMappers.remove(mapper);
 
-            // include additional mappings if we're on the from side of the mapping
-            // as the mapping may not be bi-directional
-
-            if (!mapper.hasWildcard() && name.equals(mapper.getFrom())) {
-                // this is not a wildcard value, but may map to wildcards
-                // the current example is something like log-level=wildcardCat1:level,wildcardCat2:level
-                var wildCard = PropertyMappers.getWildcardMappedFrom(mapper.getOption());
-                if (wildCard != null) {
-                    ConfigValue value = context.proceed(name);
-                    if (value != null && value.getValue() != null) {
-                        return Stream.concat(Stream.of(name), wildCard.getToFromWildcardTransformer(value.getValue()));
+            final PropertyMapper<?> mappedMapper = mapper.forKey(name);
+
+            // include additional mappings if we're on the from side of the mapping as the mapping may not be bi-directional
+            if (name.equals(mappedMapper.getFrom())) {
```

Comment:
Sorry one more thought. We should take this further:

```
            final PropertyMapper<?> mappedMapper = mapper.forKey(name);

            // only include additional mappings if we're on the from side of the mapping as the mapping may not be bi-directional
            if (!name.equals(mappedMapper.getFrom())) {
              return Stream.of(name);
            }

            allMappers.remove(mapper);

            ...
```

Reply:
@shawkins Yep, LGTM, thanks!